gdk/common/RecipeValidator.py

```diff
@@ -49,6 +52,33 @@ def __init__(self, recipe_source):
         self.recipe_source = recipe_source
         self.recipe_data = self._convert_keys_to_lowercase(self._load_recipe())
 
+    def validate_recipe_format_version(self):
+        """
+        Validate the Recipe Format Version in the recipe data.
+
+        This method checks whether the provided Recipe Format Version is valid and supported.
+
+        Raises
+        ------
+        Exception
+            If the RecipeFormatVersion field is missing or unsupported.
+
+        """
+        recipe_format_version = self.recipe_data.get("recipeformatversion")
+        if not recipe_format_version:
+            logging.error("Recipe validation failed for 'RecipeFormatVersion'. This field is required but missing "
+                          "from the recipe. Please correct it and try again.")
+            raise Exception("The recipe file is invalid. The 'RecipeFormatVersion' field is mandatory in the recipe.")
+
+        supported_recipe_version = ["2020-01-25"]
+        if recipe_format_version not in supported_recipe_version:
+            logging.error(f"Recipe validation failed for: 'RecipeFormatVersion: {recipe_format_version}'.")
+            logging.error(f"The provided RecipeFormatVersion '{recipe_format_version}' is not supported in this gdk "
```

Comment:
I'm thinking this might be better off as a warning?

Because depending on the changes to the recipe format version, it might not break an older version of gdk being able to work with it. This leaves the option for the customer to continue using their current gdk without upgrading if they want to take that risk. 

However I wouldn't want a customer to miss the warning and be surprised when something isn't working as expected later on. So if we do go the warning route maybe we can make the message extra noticeable somehow; potentially with some ascii art (if that fits in with GDK/GG design).

I'm open to hearing more opinions on this, I can see tradeoffs either way.